cfgmgr/portmgr.cpp

```diff
@@ -91,23 +95,44 @@ void PortMgr::doTask(Consumer &consumer)
                 continue;
             }
 
+            string admin_status, mtu;
+
+            bool configured = (m_portList.find(alias) != m_portList.end());
+
+            /* If this is the first time we set port settings
+             * asign default admin status and mtu
+             */
+            if (!configured)
+            {
+                admin_status = DEFAULT_ADMIN_STATUS_STR;
+                mtu = DEFAULT_MTU_STR;
+
+                m_portList.insert(alias);
+            }
+
             for (auto i : kfvFieldsValues(t))
             {
                 if (fvField(i) == "mtu")
                 {
-                    auto mtu = fvValue(i);
-                    setPortMtu(alias, mtu);
-                    SWSS_LOG_NOTICE("Configure %s MTU to %s",
-                                    alias.c_str(), mtu.c_str());
+                    mtu = fvValue(i);
                 }
                 else if (fvField(i) == "admin_status")
                 {
-                    auto status = fvValue(i);
-                    setPortAdminStatus(alias, status == "up");
-                    SWSS_LOG_NOTICE("Configure %s %s",
-                            alias.c_str(), status.c_str());
+                    admin_status = fvValue(i);
                 }
             }
+
+            if (!mtu.empty())
+            {
+                setPortMtu(alias, mtu);
+                SWSS_LOG_NOTICE("Configure %s MTU to %s", alias.c_str(), mtu.c_str());
+            }
+
+            if (!admin_status.empty())
+            {
+                setPortAdminStatus(alias, admin_status == "up");
+                SWSS_LOG_NOTICE("Configure %s %s", alias.c_str(), admin_status.c_str());
```

Comment:
can you please specify "admin status" in the log output

Reply:
Done

Reply:
Done